### ISSUE TITLE
Non-entry point methods in contracts

### ISSUE THREAD

**Issue by kevaundray** [noir-lang/noir#2688]
This is a retroactive issue.

## Problem 

Consider the following method:

```
contract Foo {
    
    struct PlaceholderStruct{x : u32 }

    fn utility_method(_context : &mut PlaceholderStruct) {}
}
```

Since entry point functions cannot contain references, the following method will fail.

## Discussion

One solution is to have a way to specify some methods as non-entry point methods. There would be two ways to define a non-entry point function, since if you pull in a Noir library, its method will be seen as non-entrypoint. There are other considerations here related to whether a function should by default be an entry point.

Another solution is to say that all methods in a contract are entry point functions and if you want a non-entry point function, you need to define it outside of the contract scope.

**Comment by TomAFrench** [comment]
We have non-entrypoint contract functions today right? @jfecher

**Comment by jfecher** [comment]
@TomAFrench yes, https://github.com/noir-lang/noir/pull/2687 added `#[contract_library_method]`
